### DISCUSSION TITLE
Dropdown - Keyboard letter scrolls to focus first available

### DISCUSSION THREAD

**Discussion by RAN-Vestjylland** [BlazorFluentUI/BlazorFluentUI#312] [Q&A]
Hi all :-)

I know this is not a functionality that is on the real React library - Having said that it would be really nice if a developer could enable some kind of "filter as you type"-thing on the dropdown or at least type in the keyboard key V to make the list scroll to for instance Volkswagen (if it is a list of car fabricants).

The javascript framework Select2 https://select2.org/dropdown shows it VERY nicely and I absolutely love all the functionality that gives. BlazorFluentUI already have most of the functionality with the grouping and headline in the dropdown items and the list, but that extra UMPH would be really amazing to have added.

I hope you guys consider to add at least the keyboard-scroll-to-focus functionality :-)

Thank you :-)

**Comment by limefrogyank** [comment]
It's that the difference between a combobox and a dropdown?  The combobox allows typing to enter the value...

So, it would just be making a combobox that filters values as you type.

**Reply by vnbaaij** [reply]
> So, it would just be making a combobox that filters values as you type.

...where you don't see that you type them...

**Reply by RAN-Vestjylland** [reply]
The dropdown is scrolling to the item where the first letter of the item is what you typed on the keyboard. So for instance if you have a dropdown with following values { "Acerola", "Apple", "Apricot", "Avocado", "Banana", "Blackberries", "Blueberries", "Breadfruit", "Date", "Durian", "Grapefuit", "Grapes", "Guava", etc. }
If you hit 'D' on the key, "Date" will be highlighted. If you hit 'D' again, "Durian" will be highlighted. If you then hit 'B' (after 'D'), "Banana" will be highlighted and so on. This is when the dropdown is readonly. (I'm comparing to WinForms dropdown or combobox.) If it's not readonly, then you can actually type in the dropdown and then the dropdown will react exactly like the Select2 example I linked in the original post (which allows filtering like mentioned above).

**Reply by vnbaaij** [reply]
I understand (and appreciate) the functionality but the dropdown as it is here (and also the React one) do not accept/handle typing of letters. Only arrow-keys (u/down), space (select) and enter (open dropdown) are accepted. Here a dropdown is always readonly.

**Reply by RAN-Vestjylland** [reply]
My point is that I would not be the user of a system where I have a dropdown with a possible item list where you have to scroll for days to find what you need. An example could be a list of car fabricants, models and variants. It will be an absolute nightmare to browse through if you don't have a quick letter scroll inside the dropdown.
I understand that this UI framework is doing what the React UI framework is doing. But nobody says you are not allowed to make it more userfriendly to actually use.
If you browse through a list of car fabricants and you know you need to select Volkswagen and hit keyboard V to go to that and then hit enter to actually select it. That is 3 keystrokes instead of hitting enter to open the list, use the keyboard arrow down and might not on first try, select Volkswagen.. Then you have to use arrow up key and maybe you missed Volkswagen again but finally you do select Volkswagen and then hit enter to select it. It just takes too long time. With my suggestion: Enter, V, Enter --> done.

**Reply by vnbaaij** [reply]
Again, I understand the functionality. But that is not what the Dropdown is for **in this library**. 
You should not use this component if you have a list with a gazillion options. We have for example the SearchBox that you can use in such a situation. That one does exactly what you were describing: 
![image](https://user-images.githubusercontent.com/1761079/118648964-3d5e9500-b7e3-11eb-8dff-15b6c05e1441.png)
Available in both single and multi select variants.
The Dropdown could be extended with the functionality you describe, but we have no way of giving feedback to the user (besides jumping to the entry) on what filtering is being applied (because the box does not allow free text input)

**Reply by RAN-Vestjylland** [reply]
ok. I get you :-)

Please try to go to the search box page and type in Aa and then use arrow keys to select an item that shows up (randomly throws out the focus of the box). I'm not sure if the React component has the functionality to use the arrow key to scroll the selection in the list that appears after it opens.

The problem with a search box is that you can't open up the list of items to select something with a mouse. A combination of SearchBox and Dropdown would be completely amazing.
